patches/server/0887-Add-option-to-hide-entities-by-default.-This-option-.patch

```diff
@@ -0,0 +1,104 @@
+     }
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 84fce7dccf9232209f939a32acfc3131e62eb27c..6016184697fbcc63ecfeb088aff63e12993bb626 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -326,6 +326,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     public void inactiveTick() { }
+     // Spigot end
+     // Paper start
++    private boolean fullyHidden = false;
+     public long activatedImmunityTick = Integer.MIN_VALUE; // Paper
+     public boolean isTemporarilyActive = false; // Paper
+     public boolean fromNetherPortal; // Paper
+@@ -337,6 +338,14 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     private UUID originWorld;
+     public boolean freezeLocked = false; // Paper - Freeze Tick Lock API
+ 
++   public void setFullyHidden(boolean hide){
++       this.fullyHidden = hide;
++   }
++
++   public boolean isFullyHidden(){
++       return this.fullyHidden;
++   }
++
+     public void setOrigin(@javax.annotation.Nonnull Location location) {
+         this.origin = location.toVector();
+         this.originWorld = location.getWorld().getUID();
+@@ -1503,7 +1512,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     }
+ 
+     public boolean isSilent() {
+-        return (Boolean) this.entityData.get(Entity.DATA_SILENT);
++        return (Boolean) this.entityData.get(Entity.DATA_SILENT) || this.isFullyHidden();  // Paper - don't play any sounds if this entity is fully hidden
+     }
+ 
+     public void setSilent(boolean silent) {
+@@ -3605,7 +3614,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+     }
+ 
+     public boolean broadcastToPlayer(ServerPlayer spectator) {
+-        return true;
++        return !this.fullyHidden; // Paper - only broadcast to a player if this entity is shown by default
+     }
+ 
+     @Override
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+index a92755211e3d42934b5efaa3f201c6c19ab7d2b4..33739749d646806b49ac247c0502b449cb7c98ff 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+@@ -1230,6 +1230,17 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
+     // Spigot end
+ 
+     // Paper start
++
++    @Override
++    public boolean isFullyHidden() {
++        return getHandle().isFullyHidden();
++    }
++
++    @Override
++    public void setFullyHidden(boolean hide) {
++        getHandle().setFullyHidden(true);
```

Comment:
The parameter hide is not used here so we can't revert back the hidden status to false

Reply:
Goot catch